bot/moderation/moderation.py

```diff
@@ -208,6 +256,52 @@ async def change_modmail_status(self, modmail: discord.Message, emoji: str, reac
 
         return discord.Embed.from_dict(dict_embed)
 
+    async def _send_confirmation_dialog(self, ctx: commands.Context, embed: discord.Embed) -> bool:
+        """Posts a confirmation dialog and returns the users answer.
+
+        Posts an embed and adds reactions for confirmation and cancellation to it. A bool indicating if the user has
+        confirmed or aborted the operation will be returned. Regardless of the return value the embed will be deleted
+        shortly after.
+
+        Args:
+            ctx (commands.Context): The context of invocation. Used to send the message.
+            embed (discord.Embed): The embed that will be posted. It should contain some explanation about what should
+            be confirmed.
+
+        Returns:
+            (bool):A bool representing the users decision.
+        """
+        message = await ctx.send(embed=embed, delete_after=constants.TIMEOUT_USER_SELECTION)
+        await message.add_reaction(constants.EMOJI_CONFIRM)
+        await message.add_reaction(constants.EMOJI_CANCEL)
+
+        def check_reaction(_reaction, user):
+            return user == ctx.author and \
+                   str(_reaction.emoji) in [constants.EMOJI_CANCEL, constants.EMOJI_CONFIRM]
+
+        reaction = await self.bot.wait_for('reaction_add', timeout=constants.TIMEOUT_USER_SELECTION, check=check_reaction)
+        await message.delete()
+
+        return str(reaction[0].emoji) == constants.EMOJI_CONFIRM
```

Comment:
What happens here if it times out?
Shouldn't reaction be none then?
`reaction[0]` would then throw an Error, right?

Reply:
A Timeout error will be raised and the execution of this code aborted.